crates/live_kit_client/src/live_kit_client.rs

```diff
@@ -1,19 +1,51 @@
 use std::sync::Arc;
 
-#[cfg(all(target_os = "macos", not(any(test, feature = "test-support"))))]
-pub mod prod;
+// #[cfg(all(
+//     target_os = "macos",
+//     not(any(test, feature = "test-support")),
+//     not(feature = "livekit-rust-sdk")
+// ))]
+// pub mod prod;
 
-#[cfg(all(target_os = "macos", not(any(test, feature = "test-support"))))]
-pub use prod::*;
+// #[cfg(all(
+//     target_os = "macos",
+//     not(any(test, feature = "test-support")),
+//     not(feature = "livekit-rust-sdk")
+// ))]
+// pub use prod::*;
 
-#[cfg(any(test, feature = "test-support", not(target_os = "macos")))]
-pub mod test;
+// #[cfg(any(test, feature = "test-support", not(target_os = "macos")))]
+// pub mod test;
 
-#[cfg(any(test, feature = "test-support", not(target_os = "macos")))]
-pub use test::*;
+// #[cfg(any(test, feature = "test-support", not(target_os = "macos")))]
+// pub use test::*;
+
+#[cfg(all(
+    target_os = "macos",
+    not(any(test, feature = "test-support")),
+    feature = "livekit-rust-sdk"
+))]
+pub mod rust;
+
+#[cfg(all(
+    target_os = "macos",
+    not(any(test, feature = "test-support")),
+    feature = "livekit-rust-sdk"
+))]
+pub use rust::*;
```

Comment:
I know this is still a work in progress, but this configuration disables the previously working builds on non-MacOS platforms added in #6859

Reply:
I appreciate the feedback but this is just to make working with Rust Analyzer easier, I'll make sure to fix everything once we're closer to shipping something like this :)